### ISSUE TITLE
Stop geolocation 127.0.0.1

### ISSUE THREAD

**Issue by kkmuffme** [woocommerce/woocommerce#52360]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

`WC_Geolocation` needs some sanity checks to not geolocate 127.0.0.1, 0.0.0.0 and 192.168.0.1 since these requests to the API are pointless, but still run e.g. on CLI requests or when developing.

Requesting `ipinfo.io/127.0.0.1/json` is pointless

### Expected behavior

No useless requests

### Actual behavior

Requests `ipinfo.io/127.0.0.1/json`

### Steps to reproduce

None

### WordPress Environment

WC 9.3.3

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by samrat-biswas-git** [comment]
Thanks @kkmuffme — valid and still applies on trunk. WC_Geolocation::geolocate_ip() has no guard for non-public IPs before falling through to geolocate_via_api(), so loopback/private/reserved addresses (127.0.0.1, 0.0.0.0, 192.168.x, etc.) still trigger pointless remote lookups (ipinfo.io/127.0.0.1/json), which is what you hit on CLI/dev requests.

Marking ready for engineering — small, additive, and backwards-compatible (it only skips lookups for IPs that would never geolocate anyway, so no change for real traffic). The fix is essentially a guard before the API fallback:

if ( ! filter_var( $ip_address, FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE ) ) {
    return ''; // private/reserved/loopback — skip the remote lookup
}

**Comment by gigitux** [comment]
I agree that geolocating local IP addresses is unnecessary, but I don’t think the impact is significant in a development environment. Keeping the request also makes development benchmarks more representative of production behavior. I’m going to close this issue for now.
